### ISSUE TITLE
pipelines take a list of torch generators but schedulers only accept a single generator

### ISSUE THREAD

**Issue by williamberman** [huggingface/diffusers#1909]
### Describe the bug

We have code to handle a list of generators in pipeline here 

https://github.com/huggingface/diffusers/blob/d67c30512048e667cbf10c298a04c73f4eb5027c/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L392-L419

However, the `generator` argument is passed directly to the scheduler which expects a single generator here

https://github.com/huggingface/diffusers/blob/d67c30512048e667cbf10c298a04c73f4eb5027c/src/diffusers/schedulers/scheduling_ddim.py#L309-L329

This currently passes the tests[^1] because eta is set to 0 so no noise is generated.

[^1]: At least the single test I'm looking at which is stable diffusion `test_inference_batch_single_identical` 

### Reproduction

Changing the scheduler to DDPM in the stable diffusion pipeline tests causes a panic. The DDPM scheduler doesn't take an eta argument and will try to generate variance noise from the generator list. 

### System Info

- `diffusers` version: 0.12.0.dev0
- Platform: Linux-5.15.0-1025-gcp-x86_64-with-glibc2.35
- Python version: 3.9.13
- PyTorch version (GPU?): 1.13.0+cu117 (True)
- Huggingface_hub version: 0.11.1
- Transformers version: 0.15.0
- Accelerate version: not installed
- xFormers version: not installed
- Using GPU in script?: A100
- Using distributed or parallel set-up in script?: no

**Comment by patrickvonplaten** [comment]
Hey @williamberman,

Thanks for opening the issue, could you maybe also post a 7 liner reproducible code snippet that we can copy-and-paste into a terminal to verify? :-)

**Comment by carson-katri** [comment]
I can reproduce with the following script:
```python
from diffusers import StableDiffusionPipeline, EulerDiscreteScheduler
import torch

pipe = StableDiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-2-1")
pipe.scheduler = EulerDiscreteScheduler.from_config(pipe.scheduler.config)
generators = [torch.Generator(device="cpu").manual_seed(1234), torch.Generator(device="cpu").manual_seed(4321)]
for i, image in enumerate(pipe(["cat digital art"] * len(generators), generator=generators).images):
    image.save(f'{i}.png')
```

**Comment by patrickvonplaten** [comment]
I think with the recent changes this issue is solved: https://github.com/huggingface/diffusers/blob/135567f18edc0bf02d515d5c76cc736d1ebddad3/src/diffusers/schedulers/scheduling_ddim.py#L326 

Also for future reference, the bug existed previously with:

```python
from diffusers import StableDiffusionPipeline, DDIMScheduler
import torch

pipe = StableDiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-2-1", torch_dtype=torch.float16)
pipe.to("cuda")
pipe.enable_attention_slicing()
pipe.scheduler = DDIMScheduler.from_config(pipe.scheduler.config)
generators = torch.Generator(device="cpu").manual_seed(1234)
generators = [torch.Generator(device="cpu").manual_seed(1234), torch.Generator(device="cpu").manual_seed(4321)]
for i, image in enumerate(pipe(2 * ["cat digital art"], generator=generators, eta=0.1, num_inference_steps=2).images):
    image.save(f'{i}.png')
```

but this code snippet works as of now.
